codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/restxml/RestXmlStructDecodeXMLGenerator.kt

```diff
@@ -0,0 +1,68 @@
+package software.amazon.smithy.aws.swift.codegen.restxml
+
+import software.amazon.smithy.aws.traits.customizations.S3UnwrappedXmlOutputTrait
+import software.amazon.smithy.model.shapes.BlobShape
+import software.amazon.smithy.model.shapes.CollectionShape
+import software.amazon.smithy.model.shapes.MapShape
+import software.amazon.smithy.model.shapes.MemberShape
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.shapes.TimestampShape
+import software.amazon.smithy.model.traits.TimestampFormatTrait
+import software.amazon.smithy.swift.codegen.SwiftTypes
+import software.amazon.smithy.swift.codegen.SwiftWriter
+import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
+import software.amazon.smithy.swift.codegen.integration.serde.xml.StructDecodeXMLGenerator
+import software.amazon.smithy.swift.codegen.model.ShapeMetadata
+import software.amazon.smithy.swift.codegen.model.hasTrait
+
+class RestXmlStructDecodeXMLGenerator(
+    val ctx: ProtocolGenerator.GenerationContext,
+    private val members: List<MemberShape>,
+    private val metadata: Map<ShapeMetadata, Any>,
+    private val writer: SwiftWriter,
+    defaultTimestampFormat: TimestampFormatTrait.Format
+) : StructDecodeXMLGenerator(ctx, members, metadata, writer, defaultTimestampFormat) {
+    override fun render() {
+        writer.openBlock("public init (from decoder: \$N) throws {", "}", SwiftTypes.Decoder) {
+            if (members.isNotEmpty()) {
+                renderDecodeBody()
+            }
+        }
+    }
+
+    private fun renderDecodeBody() {
+        val containerName = "containerValues"
+        if (responseBodyIsNotWrapped()) {
+            writer.write("var $containerName = try decoder.unkeyedContainer()")
+            members.forEach { member ->
+                renderSingleMemberUnkeyed(member, containerName)
+            }
+        } else {
+            writer.write("let $containerName = try decoder.container(keyedBy: CodingKeys.self)")
+            members.forEach { member ->
+                renderSingleMember(member, containerName)
+            }
+        }
+    }
+    private fun responseBodyIsNotWrapped(): Boolean {
+        if (metadata.containsKey(ShapeMetadata.OPERATION_SHAPE)) {
+            val operationShape = metadata[ShapeMetadata.OPERATION_SHAPE] as OperationShape
+            if (operationShape.hasTrait<S3UnwrappedXmlOutputTrait>()) {
```

Comment:
since this only applies to s3, should this be a customization to the decode xml generator or in here?

Reply:
Are you suggesting a customization via an integration?  If so, that's an interesting thought.. but didn't consider it yet because it doesn't seem like there is much risk in other restxml protocols using the `S3UnwrappedOutputTrait`

Reply:
Ya it just seems to weird to call out an s3 specific trait here in a class where it's more general to all rest xml services. Even though it's not affecting them. Just feels a little off but I also trust your judgment if you wanted to leave it here

Reply:
Hmmm -- Let me try and implement it as a section writer -- if it is cleaner, then we'll go with that, but at least we can have something to compare it with.

Reply:
Looked into implementing this with an integration/section writer, and does not seem to be the right tool for the job.  Why? because section writers are really good at writing/generating code (not passing back a value) to the original context.  For example, if we wanted to put the following code in an integration only enabled for S3:
```
    private fun responseBodyIsNotWrapped(): Boolean {
        if (metadata.containsKey(ShapeMetadata.OPERATION_SHAPE)) {
            val operationShape = metadata[ShapeMetadata.OPERATION_SHAPE] as OperationShape
            if (operationShape.hasTrait<S3UnwrappedXmlOutputTrait>()) {
                return true
            }
        }
        return false
    }
```

Then a section writer is not the right tool for this -- we'd have to do something like modify the `SwiftIntegration` protocol in a way that we could call into this customization.

So, in that regard -- if we wanted to make `RestXmlStructDecodeXMLGenerator` generic, we could use the SwiftIntegration's `preprocessModel()` in order to detect whether or not an operation has `S3UnwrappedXmlOutputTrait` applied to it, and then replace it with a newly created trait like `UnwrappedXmlOutputTrait` -- so that way the code inside of `RestXmlStructDecodeXMLGenerator` becomes:
```
    private fun responseBodyIsNotWrapped(): Boolean {
        if (metadata.containsKey(ShapeMetadata.OPERATION_SHAPE)) {
            val operationShape = metadata[ShapeMetadata.OPERATION_SHAPE] as OperationShape
            if (operationShape.hasTrait<UnwrappedXmlOutputTrait>()) {
                return true
            }
        }
        return false
    }
```

This doesn't seem like a huge win, and seems to overcomplicate the code generation, so seems like it is reasonable just to leave it until we have a stronger case for changing it.